server/opts.go

```diff
@@ -3185,14 +3185,11 @@ func ConfigureOptions(fs *flag.FlagSet, args []string, printVersion, printHelp,
 			if opts.CheckConfig {
 				return nil, err
 			}
-
-			// If only warnings then can still continue.
-			if cerr, ok := err.(*processConfigErr); ok && len(cerr.Errors()) == 0 {
-				fmt.Fprint(os.Stderr, err)
-				return opts, nil
+			if cerr, ok := err.(*processConfigErr); !ok || len(cerr.Errors()) != 0 {
+				return nil, err
 			}
-
-			return nil, err
+			// If we get here we only have warnings and can still continue
+			fmt.Fprint(os.Stderr, err)
```

Comment:
Are we not changing the logic here? Code in 2.1.4 is:
```
// If only warnings then can still continue.
if cerr, ok := err.(*processConfigErr); ok && len(cerr.Errors()) == 0 {
    fmt.Fprint(os.Stderr, err)
    return opts, nil
}
return nil, err
```
which means that if it was an error (but only warnings), then we print and stop by returning `opts, nil` but with your changes we return the error if there is error, but if only warning then we continue. I am not saying this is wrong, but maybe @wallyqs can comment if this was the actual original intent?

Reply:
We want to treat having warnings the same way as having no error + print the waring.
The problem resulted from having a warning, then returning opts, nil WITHOUT having executed later code  fs.Parse(args), line 3201.

This change returns an error if we deal with a different error type or if the error contains at least one of our errors.
Otherwise, we print the warning, and continue as if there was no error to begin with.

Reply:
gotcha.. so the bug was actually from before and not introduced by one of your recent change with config type assertion handling.

Reply:
Yes that was from before. Any warning triggers this. 
(The assertion handling changes would result in an error)